src/charmed_kubeflow_chisme/testing/cos_integration.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Utilities for testing COS integration with charms."""
+import logging
+from pathlib import Path
+from typing import Any, Dict, Set
+
+import yaml
+from juju.action import Action
+from juju.application import Application
+from juju.model import Model
+from juju.unit import Unit
+
+log = logging.getLogger(__name__)
+
+GRAFANA_AGENT_APP = "grafana-agent-k8s"
+METRICS_ENDPOINT_RELATION = "metrics-endpoint"
+GRAFANA_DASHBOARD_RELATION = "grafana-dashboard"
+ALER_RULES_DIRECTORY = Path("./src/prometheus_alert_rules")
+
+
+async def deploy_and_assert_grafana_agent(
+    model: Model,
+    app: str,
+    channel: str = "latest/stable",
+    dashboard: bool = True,
+    metrics: bool = True,
+) -> None:
+    """Deploy grafana-agent-k8 and add relate it with app.
+
+    Helper function to deploy and relate grafana-agent-k8s with provided app.
+
+    Args:
+        model: libjuju model object
+        app: name of application with which the Grafana agent should be related
+        channel: grafana-agent-k8s channel name, latest/stable as default
+        dashboard: relate <app>:grafana-dashboard grafana-agent-k8s:grafana-dashboards-consumer
+        metrics: relate <app>:metrics-endpoint grafana-agent-k8s:metrics-endpoint
+    """
+    assert app in model.applications, f"application {app} was not found"
+
+    log.info("deploying %s from %s channel", GRAFANA_AGENT_APP, channel)
+    await model.deploy(GRAFANA_AGENT_APP, channel=channel)
+
+    if dashboard is True:
+        log.info(
+            "juju relate %s:%s %s:grafana-dashboards-consumer",
+            app,
+            GRAFANA_DASHBOARD_RELATION,
+            GRAFANA_AGENT_APP,
+        )
+        await model.add_relation(
+            f"{app}:{GRAFANA_DASHBOARD_RELATION}",
+            f"{GRAFANA_AGENT_APP}:grafana-dashboards-consumer",
+        )
+
+    if metrics is True:
+        log.info(
+            "juju relate %s:%s %s:metrics-endpoint",
+            app,
+            METRICS_ENDPOINT_RELATION,
+            GRAFANA_AGENT_APP,
+        )
+        await model.add_relation(
+            f"{app}:{METRICS_ENDPOINT_RELATION}", f"{GRAFANA_AGENT_APP}:metrics-endpoint"
+        )
+
+    await model.wait_for_idle(apps=[GRAFANA_AGENT_APP], status="blocked", timeout=5 * 60)
```

Comment:
The [wait_for_idle](https://github.com/juju/python-libjuju/blob/d4f667766655e1eadbb5fd8e6cb40eb9e9451039/juju/model.py#L2791) is not supporting checking status message, but we can add simple `assert` after this one. I think it would be useful :+1:.
Something like:
```python
assert model.applications[GRAFANA_AGENT_APP].status_message == "..."
```